crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -715,13 +716,23 @@ impl<'w, 's, 'a> EntityCommands<'w, 's, 'a> {
     /// # bevy_ecs::system::assert_is_system(add_combat_stats_system);
     /// ```
     pub fn insert(&mut self, bundle: impl Bundle) -> &mut Self {
-        self.commands.add(Insert {
+        self.add(Insert {
             entity: self.entity,
             bundle,
         });
```

Comment:
If we're changing `Insert` to implement `EntityCommand` instead of `Command`, then we don't need to store the entity as a field anymore. It gets passed as an argument when `EntityCommand::apply` gets called.